src/provider/rsa_keymgmt.c

```diff
@@ -1203,6 +1203,16 @@ static void *ibmca_keymgmt_rsa_gen(void *vgenctx, OSSL_CALLBACK *osslcb,
         }
     }
 
+    /* If p < q, swap and recalcualte now */
+    rc = ica_rsa_crt_key_check(&key->rsa.private);
+    if (rc > 1) {
+        rc = 0;
```

Comment:
I think it's not necessary to set rc = 0 because you are returning from the routine anyway here.

Reply:
right, removed.